### ISSUE TITLE
Issue with portrait images with EXIF orientation tag set

### ISSUE THREAD

**Issue by ervee** [robertvandenbaar/LDG#4]
Hi,

I think there's an issue with the EXIF orientation processing. Well, first of all I think the EXIF orientation tag is the worst invention since the beginning of man-kind, but we'll have to live with it I think.

I normally lossless rotate my images to the correct orientation and correct or remove the  EXIF orientation tag, but I noticed I still have some pictures shot in portrait without this post-processing and they still have the EXIF orientation tag set.

These pictures show:

- In landscape (wrong) in LDG thumbnails, resized and full-size view.
- In portrait (correct) when I right-mouse > View Image the resized and full-size image from LDG in Firefox 53
- In landscape when the thumbnail it viewed like that 🤕
- In landscape in Internet Explorer in every picture as expected from IE...
- In portrait in proper editing software (and not set to ignore the EXIF orientation tag)
- In landscape in Windows Photo Viewer/Preview thing. Which is expected, it does not support the EXIF orientation tag. This is fixed in Windows 8 and upward if I'm correct.

So long story short: Is there a way to have LDG support the EXIF orientation tag :)

**Comment by robertvandenbaar** [comment]
In November 2016 I added some EXIF functionality. This code auto-rotates the images based on the EXIF value and then strips the EXIF data (so it will always be displayed correctly regardless of browser). Later on I removed the auto-rotate option when GD is being used (because it didn't work correctly).

I you haven't already, you can enable AUTO_ROTATE in config.php. This will auto-rotate (like described above) the thumbnail and medium sized image. Not the original of course (so viewing in original mode can still result in incorrectly orientated image).

**Comment by ervee** [comment]
So, I finally remembered to try your suggestion :)
I added this to config.php
```
/**
 * Auto rotate images based on EXIF tag
 * NOTE: does NOT work good with GD
 */
define('AUTO_ROTATE', true);
```
Then deleted the cache/detail and cache/thumbnail folders.
Then refreshed the browser which was in the gallery showing an image folder which had a lot of wrongly oriented images.

The images still show incorrectly :(

**Comment by robertvandenbaar** [comment]
Did you also update to the latest version of LDG?

**Comment by ervee** [comment]
I guess I did not :) I just moved the old web-folder, unzipped a new download from GitHub and reconfigured the config.php.

Because all cache is now gone it is generating new previews etc. But it throws a "...failed to generate..." error and waits forever in every folder I check (even with auto rotate false!). The new thumbnail cache files are generated though and when I refresh the page, the thumbnails do show! Weird...

I'm now using service.resize.images.php to populate the cache. This throws 2 errors to start with:
```
PHP Warning:  session_start(): Cannot send session cookie - headers already sent by (output started at /srv/www/vhosts/foto.moskovic.org/htdocs/baarrgal/config.php:38) in /srv/www/vhosts/foto.moskovic.org/htdocs/baarrgal/init.php on line 20
PHP Warning:  session_start(): Cannot send session cache limiter - headers already sent (output started at /srv/www/vhosts/foto.moskovic.org/htdocs/baarrgal/config.php:38) in /srv/www/vhosts/foto.moskovic.org/htdocs/baarrgal/init.php on line 20
```
It is still running...

**Comment by robertvandenbaar** [comment]
I think you have some whitespace (or other text) after the PHP closing tag in config.php

**Comment by ervee** [comment]
Yes, 1 blank line. I removed it and it all looks good now!
